composeApp/build.gradle.kts

```
@@ -80,3 +88,34 @@ dependencies {
     debugImplementation(compose.uiTooling)
 }
 
+val generateBuildConfig by tasks.registering {
+    val outputDir = layout.buildDirectory.dir("generated/buildConfig")
+
+    val props = Properties().apply {
+        load(rootProject.file("local.properties").inputStream())
+    }
+
+    val baseUrl = props["BASE_URL"] ?: "https://test/"
```

I think the Test URL is arbitrary; maybe if it throws an exception, it will be better!
```suggestion
    val baseUrl = props["BASE_URL"] ?: ?: throw IllegalStateException("BASE_URL not found in local.properties")
```